### PR TITLE
fix(repository): findPath returns null without data received 

### PR DESCRIPTION
**Pull request by doktordirk**

### CODE REVIEW

File: src/repository.js [hunk 1]
```diff
@@ -105,7 +105,7 @@ export class Repository {
     }
 
     return findQuery
-      .then(x => this.populateEntities(x))
+      .then(x => x ? this.populateEntities(x) : Promise.reject(new Error('No data')))
```
Comment (RWOverdijk) [inline src/repository.js]: You can just throw, no need to Promise.reject.

**Comment by RWOverdijk** [comment]
The server returns an empty array. Never an empty response, right? If it's an empty response, JSON.parse() would throw. If it's an empty array or null, I still don't think we should just throw an exception. That's up to the developer implementing this.

**Comment by doktordirk** [comment]
for x =[] -> !!x === true. so, it empty arrays would handed over to populateEntities and JSON parsed
for x= null, undefined -> !!x === false, don't parse (would throw obscure error), but throw meaningful error now (that's the idea at least)

File: src/repository.js (hunk 1 above)
Comment (doktordirk) [reply src/repository.js]: i always get that confused. but right i think rejection with non-error was what one shouldn't do

Comment (doktordirk) [reply src/repository.js]: what do you prefer
`.then(x= > x ? this.populateEntities(x) : Promise.reject(new Error('No data')))`
or

``` js
.then(x= > {
  if (!x) {
   throw new Error('No data');
  }
  return this.populateEntities(x);
}
```

i prefer the former

Comment (RWOverdijk) [reply src/repository.js]: I prefer the latter, as well as changing `x` to something more verbose.

**Comment by RWOverdijk** [comment]
I get that, but null doesn't have to throw an error. You can get a 201 with an empty response, which shouldn't throw an error. I don't agree with this.

**Comment by doktordirk** [comment]
basically Exactly! currently that would throw a JSON.parse error down the line. i prefer fail fast and hard :smile:

**Comment by doktordirk** [comment]
sorry, not JSON.parse error, but an error from populated.markClean(); down a few lines, as populated is null
so the error thrown currently is 'populated.markClean() is not a function/undefined or something'. which isn't a helpfull error message

**Comment by RWOverdijk** [comment]
👍

**Review by RWOverdijk — APPROVED** [review]

**Pull request merged**